src/platform/i386/miniacpi.c

```diff
@@ -0,0 +1,108 @@
+#include "kernel.h"
+#include "string.h"
+#include "mem_layout.h"
+#include "pgtbl.h"
+
+typedef struct {
+	char signature[8];
+	u8_t checksum;
+	char oemid[6];
+	u8_t revision;
+	u32_t rsdtaddress;
+	u32_t length;
+	u64_t xsdtaddress;
+	u8_t extendedchecksum;
+	u8_t reserved[3];
+} __attribute__((packed)) RSDP;
+
+typedef struct rsdt {
+	char signature[4];
+	u32_t length;
+	u8_t revision;
+	u8_t checksum;
+	char oemid[6];
+	char oemtableid[8];
+	u32_t oemrevision;
+	u32_t creatorid;
+	u32_t creatorrevision;
+	struct rsdt *entry[0];
+} __attribute__((packed)) RSDT;
+
+extern u8_t *boot_comp_pgd;
+u32_t basepage;
+static RSDT *rsdt;
+
+static inline void *
+pa2va(void *pa)
+{
+	return (void*)(((u32_t)pa & ((1<<22)-1)) | basepage);
+}
+
+void *
+acpi_find_rsdt(void)
+{
+	unsigned char *sig;
+	RSDP *rsdp = NULL;
+	for (sig = (unsigned char*)0xc00E0000; sig < (unsigned char*)0xc00FFFFF; sig += 16) {
```

Comment:
These constants should likely be in a macro since they are reused, and for documentation (i.e. the name of the macro)